capabilities/capabilities.go

```diff
@@ -31,75 +27,15 @@ var (
 	}
 
 	// NodePools is the capabilitiy to group tenant cluster workers logically.
+	// Details get completed with API data, if the feature is available.
 	NodePools = CapabilityDefinition{
 		Name: "NodePools",
-		RequiredReleasePerProvider: []ReleaseProviderPair{
-			ReleaseProviderPair{
-				Provider:       "aws",
-				ReleaseVersion: semver.MustParse("9"), // TODO: fix once the node pools release version is defined.
-			},
-		},
 	}
 
 	// AllCapabilityDefinitions contains all the capabilities
-	AllCapabilityDefinitions = []CapabilityDefinition{
-		Autoscaling,
-		AvailabilityZones,
-		NodePools,
+	AllCapabilityDefinitions = []*CapabilityDefinition{
+		&Autoscaling,
```

Comment:
These pointers bother me a bit because they practically expose these exported instances globally and therefore any code anywhere could modify them and accidentally break logic elsewhere.

I'm thinking that there should be probably non-pointer instances of these capabilities and when they are used together with the `Service` instance, then that service should keep its internal copy with completed information such as NodePools release versions etc.

Reply:
Changed this